tfs/writer.py

```diff
@@ -147,85 +167,137 @@ def _insert_index_column(data_frame: TfsDataFrame | pd.DataFrame, save_index: st
 def _get_headers_string(headers_dict: dict, width: int) -> str:
     """
     Returns the string to write a ``TfsDataFrame`` headers to file. Will return an empty string if
-    called for an empty headers dictionary, in order not write an line to file.
+    called for an empty headers dictionary, in order not write a line to file.
 
     Args:
         headers_dict (dict): the ``TfsDataFrame`` headers.
         width (int): column width to use when formatting keys and values from the headers dict.
 
     Returns:
-        A full string representation for the headers dictionary.
+        A full string representation for the headers dictionary, TFS compliant.
     """
     if headers_dict:
         return "\n".join(_get_header_line(name, headers_dict[name], width) for name in headers_dict)
     return ""
 
 
 def _get_header_line(name: str, value, width: int) -> str:
+    """
+    Creates and returns the string value for a single header line, based
+    on the name of the header parameter and its value.
+
+    For instance, calling this for 'param' header which is a float equal to
+    1.792 and using the DEFAULT_COLUMN_WIDTH of the package would yield:
+    "@ param                %le                1.792"
+
+
+    Args:
+        name (str): name of the header parameter, as its entry in the headers
+            dictionary.
+        value: value of the header parameter. Any valid type is accepted here
+            (int, float, str, potentially bool, complex etc.) and the type
+            of this value is used to infer the formatting.
+        width (int): column width to use when formatting the header line.
+
+    Returns:
+        The full, formatted header line string.
+    """
     if not isinstance(name, str):
         errmsg = f"{name} is not a string"
         raise TypeError(errmsg)
     type_str = _value_to_type_string(value)
-    if type_str == "%s":
-        value = f'"{value}"'
-    return f"@ {name:<{width}} {type_str} {value:>{width}}"
+    value_str = TfsStringFormatter().format_field(value, _value_to_string_format_id(value))
```

Comment:
**FEATURE**

This is where the work is done. The `TfsStringFormatter` takes care of the proper string representation, based on the dtype. See implementation below in the file.